### PR TITLE
MVP for TV main activity 

### PR DESCRIPTION
**Pull request by blazsolar**

### CODE REVIEW

File: tv/src/main/java/butter/droid/tv/fragments/TVMediaGridFragment.java [hunk 1]
```diff
@@ -199,9 +199,7 @@ else if (media.getMedia() instanceof Show)
     public void onItemSelected(Presenter.ViewHolder itemViewHolder, Object item, RowPresenter.ViewHolder rowViewHolder, Row row) {
         if (item instanceof MediaCardPresenter.MediaCardItem) {
             MediaCardPresenter.MediaCardItem overviewItem = (MediaCardPresenter.MediaCardItem) item;
-//            if (overviewItem.isLoading()) return;
-
-            mBackgroundUpdater.updateBackgroundAsync(((MediaCardPresenter.MediaCardItem) item).getMedia().headerImage);
+            mBackgroundUpdater.updateBackgroundAsync(overviewItem.getMedia().headerImage);
```
Comment (aldoborrero) [inline tv/src/main/java/butter/droid/tv/fragments/TVMediaGridFragment.java]: Remove m prefix

File: tv/src/main/java/butter/droid/tv/fragments/TVBaseDetailsFragment.java [hunk 2]
```diff
@@ -46,137 +46,144 @@
...
-	public static final String EXTRA_ITEM = "item";
-	public static final String EXTRA_HERO_URL = "hero_url";
+    public static final String EXTRA_ITEM = "item";
+    public static final String EXTRA_HERO_URL = "hero_url";
 
-	private ArrayObjectAdapter mAdapter;
-	private ClassPresenterSelector mPresenterSelector;
-	private Media mItem;
-	private String mHeroImage;
+    private ArrayObjectAdapter mAdapter;
```
Comment (aldoborrero) [inline tv/src/main/java/butter/droid/tv/fragments/TVBaseDetailsFragment.java]: Let's remove mPrefix if possible

File: tv/src/main/java/butter/droid/tv/presenters/showdetail/EpisodeCardPresenter.java [hunk 3]
```diff
@@ -37,117 +37,122 @@
...
-		final Episode episode = (Episode) item;
-		((ViewHolder) viewHolder).setEpisode(episode);
-		((ViewHolder) viewHolder).getCardView().setTitleText(episode.title);
-		((ViewHolder) viewHolder).getCardView().setContentText(
-            String.format(mContext.getString(R.string.episode_number_format), episode.episode));
-		((ViewHolder) viewHolder).getCardView().setMainImageDimensions(mCardWidth, mCardHeight);
-		((ViewHolder) viewHolder).updateCardViewImage(episode.image);
+    @Override
+    public ViewHolder onCreateViewHolder(ViewGroup parent) {
+        mContext = parent.getContext();
```
Comment (aldoborrero) [inline tv/src/main/java/butter/droid/tv/presenters/showdetail/EpisodeCardPresenter.java]: Same here

File: tv/src/main/java/butter/droid/tv/fragments/TVBaseDetailsFragment.java (hunk 2 above)
Comment (blazsolar) [inline tv/src/main/java/butter/droid/tv/fragments/TVBaseDetailsFragment.java]: 👍  done

File: tv/src/main/java/butter/droid/tv/presenters/showdetail/EpisodeCardPresenter.java (hunk 3 above)
Comment (blazsolar) [inline tv/src/main/java/butter/droid/tv/presenters/showdetail/EpisodeCardPresenter.java]: also done

**Pull request merged**